packages/ui-lib/src/Form/atoms/Switch.tsx

```diff
@@ -281,7 +294,7 @@ const Switch : React.FC<IProps> = ({ state = 'default', leftTheme = 'off', right
         </SwitchInner>
       </SwitchOuter>
       {labelText ? <LabelText>{labelText}</LabelText> : null}
-      <RealInput ref={inputRef} type='checkbox' disabled={state !== 'default' && state !== 'failure'} defaultChecked={checked} />
+      <RealInput ref={inputRef} type='checkbox' disabled={state !== 'default' && state !== 'failure'} {...(isDefaultPropPassed ? { defaultChecked } : { defaultChecked: checked })} /> 
```

Comment:
The condition here although is syntactically valid it's a bit complicated to read 
{...(isDefaultPropPassed ? { defaultChecked } : { defaultChecked: checked })}

I would recommend to use the de verbose version for such occasion :

{
  isDefaultPropPassed 
    ? defaultChecked={defaultChecked} 
    : defaultChecked={checked}
}

However I think a review of how to handle the values is required since input type checkbox does have a defaultCheck property as well as a checked property.

https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/input/checkbox